### PR TITLE
add pluginfile for serving of smartmedia assets

### PR DESCRIPTION
**Pull request by mattporritt**
as well as initial checks to make sure the smartmedia file correctly matches the source file

### CODE REVIEW

File: lib.php [hunk 1]
```diff
@@ -33,3 +33,63 @@
...
+ * Serve the files from the local smartmedia file areas.
+ *
+ * @param stdClass $course The course object.
+ * @param stdClass $cm The course module object.
+ * @param stdClass $context The context.
+ * @param string $filearea The name of the file area.
+ * @param array $args Extra arguments (itemid, path).
+ * @param bool $forcedownload Whether or not force download.
+ * @param array $options Additional options affecting the file serving.
+ * @return bool False if the file not found, just send the file otherwise and do not return anything.
```
Comment (tomdickman) [inline lib.php]: Would it not be better to return `true` even if we aren't handling it at the moment, just for consistency?

File: tests/conversion_test.php [hunk 2]
```diff
@@ -847,4 +847,65 @@ public function test_get_pathnamehashes() {
...
+        $conversionrecord->rekog_moderation_status = $conversion::CONVERSION_FINISHED;
+        $conversionrecord->rekog_face_status = $conversion::CONVERSION_NOT_FOUND;
+        $conversionrecord->rekog_person_status = $conversion::CONVERSION_FINISHED;
+        $conversionrecord->timecreated = time();
+        $conversionrecord->timemodified = time();
+
+        $DB->insert_record('local_smartmedia_conv', $conversionrecord);
+
+        $result = $conversion->check_smartmedia_file($sourcefile, $smartfile);
+        $this->assertFalse($result); // Should be false as there a contenthash path mismatch.
```
Comment (tomdickman) [inline tests/conversion_test.php]: 'Should be false as there is a contenthash path mismatch.'

File: lib.php (hunk 1 above)
Comment (mattporritt) [reply lib.php]: Nope, this is old school Moodle code. The sendfile method that is called eventually calls die; when the file is sent. So the final return true will never actually be reached.

**Comment by mattporritt** [comment]
@tomdickman I've fixed the code comment in the test

**Pull request merged**